### PR TITLE
[CSM] Add function to get player privileges

### PR DESCRIPTION
**Pull request by red-001**
This also moves some related helper functions to `misc_helpers.lua`

### CODE REVIEW

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -723,3 +723,28 @@ function core.pointed_thing_to_face_pos(placer, pointed_thing)
...
+	local list = {}
+	for priv, bool in pairs(privs) do
+		if bool then
+			list[#list + 1] = priv
+		end
+	end
+	return table.concat(list, delim)
+end
+
+assert(core.string_to_privs("a,b").b == true)
```
Comment (nerzhul) [inline builtin/common/misc_helpers.lua]: nice, auto unit tests

File: src/client.h [hunk 2]
```diff
@@ -417,6 +417,7 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	bool checkPrivilege(const std::string &priv) const
 	{ return (m_privileges.count(priv) != 0); }
 
+	std::unordered_set<std::string> getPrivilegeList() const {return m_privileges;};
```
Comment (nerzhul) [inline src/client.h]: you can return const ref here

**Comment by red-001** [comment]
Fixed the code style issue

File: src/client.h [hunk 3]
```diff
@@ -417,6 +417,7 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	bool checkPrivilege(const std::string &priv) const
 	{ return (m_privileges.count(priv) != 0); }
 
+	const std::unordered_set<std::string> &getPrivilegeList() {return m_privileges;};
```
Comment (nerzhul) [inline src/client.h]: the extra ; after braces is not needed, i you can add space inside braces too can be nice :). Also this function should be const (sorry for to be late about that)

Comment (red-001) [inline src/client.h]: done

File: src/script/lua_api/l_client.cpp [hunk 4]
```diff
@@ -329,6 +329,16 @@ int ModApiClient::l_take_screenshot(lua_State *L)
 	return 0;
 }
 
+int ModApiClient::l_get_privilege_list(lua_State *L)
+{
+	Client *client = getClient(L);
```
Comment (Zeno-) [inline src/script/lua_api/l_client.cpp]: Can be const or not?

**Comment by Zeno-** [comment]
:+1: apart from my single comment

**Pull request merged**